### PR TITLE
laminas/laminas-coding-standard:^3.0 + Apply code style

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

File: phpcs.xml
Comment (fezfez) [inline phpcs.xml]: If this file is not exclude it crash

File: src/Controller/ControllerManager.php
Comment (froschdesign) [inline src/Controller/ControllerManager.php]: An error that is constantly repeated.

File: phpcs.xml
Comment (gsteel) [reply phpcs.xml]: Just checked out this PR and phpcs is fine without it for me and reporting a bunch of issues. Can you try another `composer update` ensuring you do that on at least PHP 8.1?

File: src/ApplicationInterface.php
Comment (gsteel) [inline src/ApplicationInterface.php]: Adding `declare(strict_types=1)` is a bit too risky on this lib for a minor release - can you please revert and disable the relevant rule for `src` only - having strict types enabled in `/test` is good and desirable.

File: src/Controller/ControllerManager.php
Comment (gsteel) [reply src/Controller/ControllerManager.php]: I can't remember which rule causes this problem, but IIRC you need to append the import with `// phpcs:ignore` rather than disabling it in `phpcs.xml`

File: src/Controller/ControllerManager.php
Comment (gsteel) [inline src/Controller/ControllerManager.php]: Incorrect casing here too

File: src/Controller/ControllerManager.php
Comment (gsteel) [inline src/Controller/ControllerManager.php]: And here…

File: src/Controller/ControllerManager.php
Comment (gsteel) [inline src/Controller/ControllerManager.php]: And here…

File: src/Controller/LazyControllerAbstractFactory.php
Comment (gsteel) [inline src/Controller/LazyControllerAbstractFactory.php]: Fix and re-check case throughout here

File: src/Controller/Plugin/Service/ForwardFactory.php
Comment (gsteel) [inline src/Controller/Plugin/Service/ForwardFactory.php]: Fix and re-check casing throughout this file

File: src/DispatchListener.php
Comment (gsteel) [inline src/DispatchListener.php:103]: 👍

File: src/MiddlewareListener.php
Comment (gsteel) [inline src/MiddlewareListener.php]: Bad Casing here

File: src/Service/AbstractPluginManagerFactory.php
Comment (gsteel) [inline src/Service/AbstractPluginManagerFactory.php]: Bad casing here…

File: src/Service/ApplicationFactory.php
Comment (gsteel) [inline src/Service/ApplicationFactory.php]: And again…

File: src/Service/ConfigFactory.php
Comment (gsteel) [inline src/Service/ConfigFactory.php]: I'll just keep pointing these out 😂

File: src/Service/ControllerManagerFactory.php
Comment (gsteel) [inline src/Service/ControllerManagerFactory.php]: 👆

File: src/Service/DispatchListenerFactory.php
Comment (gsteel) [inline src/Service/DispatchListenerFactory.php]: 👆

File: src/Service/EventManagerFactory.php
Comment (gsteel) [inline src/Service/EventManagerFactory.php]: OK, I'll stop now…

File: src/Service/ServiceListenerFactory.php
Comment (gsteel) [inline src/Service/ServiceListenerFactory.php]: Sorry to be a pain in the ass here, but can you instead `use Laminas` and just remove the leading `\` where used - my reasoning is that the import list gets massive in a file like this and when scanning, it's more helpful to see the namespace inline

File: src/View/Http/InjectTemplateListener.php
Comment (gsteel) [inline src/View/Http/InjectTemplateListener.php]: This would be better as `$replacement === false`

**Review by gsteel — CHANGES_REQUESTED** [review]
Thanks for all the work here @fezfez 👍

File: src/Service/ServiceListenerFactory.php
Comment (fezfez) [reply src/Service/ServiceListenerFactory.php]: sorry but this request change is verrryyy annoying :s, could you do it by yourself once it's merged ?

Comment (gsteel) [reply src/Service/ServiceListenerFactory.php]: Nope. You started this 😜

Comment (froschdesign) [reply src/Service/ServiceListenerFactory.php]: @fezfez 
I agree with @gsteel here. It would be great if you could still make this change.
Thanks in advance! 👍🏻

Comment (fezfez) [reply src/Service/ServiceListenerFactory.php]: @gsteel @froschdesign i've done something in between, let me know if it's ok for you :)

**Review by gsteel — APPROVED** [review]
This looks good to me @fezfez Thank you - I'd like another pair of eyes on it though, because it's a big patch.

cc @laminas/technical-steering-committee

**Comment by fezfez** [comment]
> This looks good to me @fezfez Thank you - I'd like another pair of eyes on it though, because it's a big patch.
> 
> cc @laminas/technical-steering-committee

Yep, let me know when it's ok, I will rebase

File: phpcs.xml
Comment (weierophinney) [inline phpcs.xml:22]: Hate this, but it's necessary to prevent BC breakage.

File: test/_autoload.php
Comment (weierophinney) [inline test/_autoload.php]: This is something phpcs does when it doesn't recognize a class, due to it not being loaded. Which ... of course it isn't, but that's why the conditional is here. While this is likely safe (class names are technically case insensitive in the engine), this should probably be reverted, and an ignore rule setup for this file.

**Review by weierophinney — CHANGES_REQUESTED** [review]
I've done a spot check, and it basically looks fine - with the caveat of the `_autoload.php` file, where phpcs was a bit aggressive in changing a non-existent class name.

File: test/_autoload.php
Comment (fezfez) [reply test/_autoload.php]: done

File: phpcs.xml
Comment (fezfez) [reply phpcs.xml:22]: yep :/

**Comment by fezfez** [comment]
> I've done a spot check, and it basically looks fine - with the caveat of the `_autoload.php` file, where phpcs was a bit aggressive in changing a non-existent class name.

@weierophinney done

**Review by gsteel — APPROVED** [review]
Thanks @fezfez and @weierophinney :)

**Pull request merged**
